ui/desktop/src/shared_sessions.ts

```diff
@@ -0,0 +1,267 @@
+                  type: 'text',
+                  text: 'Search results saved to: /Users/smohammed/.cache/goose/computer_controller/search_20250310_113104.json',
+                },
+              ],
+              status: 'success',
+            },
+            id: 'toolu_bdrk_01TmhtTTEkGpRX2kjGjoXdTk',
+          },
+        ],
+        role: 'user',
+      },
+      {
+        created: 1741620676,
+        content: [
+          {
+            type: 'text',
+            text: "Based on recent research, here's a comprehensive overview of countries with the highest income tax rates:\n\n### Countries with Highest Income Tax Rates\n\n1. **Scandinavian Countries**\n   - Denmark: Up to 55.9%\n   - Sweden: Up to 57.1%\n   - Finland: Up to 56.4%\n\n2. **Other High-Tax Countries**\n   - Japan: Up to 55.7%\n   - Belgium: Up to 53.7%\n   - Austria: Up to 55%\n   - Netherlands: Up to 52%\n\n### Comparison with Low-Tax Countries\n\n**Low-Tax Countries**:\n- United Arab Emirates: 0%\n- Bahamas: 0%\n- Bermuda: 0%\n- Cayman Islands: 0%\n- Monaco: 0%\n\n### Key Differences and Implications\n\n1. **Social Services**\n   - High-tax countries typically provide:\n     - Comprehensive healthcare\n     - Free or subsidized education\n     - Strong social safety nets\n     - Extensive public infrastructure\n\n2. **Economic Impact**\n   - High taxes can:\n     - Reduce individual disposable income\n     - Fund extensive government programs\n     - Potentially discourage high-income earners\n\n3. **Low-Tax Alternatives**\n   - Often small countries or financial centers\n   - Attract businesses and wealthy individuals\n   - May have fewer social services\n   - Rely on alternative revenue sources (tourism, financial services)\n\n### Considerations\n\n- Tax rates are not just about the percentage, but also about:\n  - How the tax revenue is used\n  - Quality of public services\n  - Overall cost of living\n  - Economic opportunities\n\nIt's worth noting that tax rates can change, and there are many nuances in tax systems, including deductions, brackets, and local variations.\n\nWould you like me to elaborate on any specific aspect of global income taxation?",
+          },
+        ],
+        role: 'assistant',
+      },
+    ],
+  };
+}
+
+/**
+ * Creates a new shared session
+ * @param baseUrl The base URL for session sharing API
+ * @param messages The messages to include in the shared session
+ * @param description Optional description for the shared session
+ * @returns Promise with the share token
+ */
+export async function createSharedSession(
+  baseUrl: string,
+  messages: Message[],
+  description?: string
+): Promise<string> {
+  try {
+    const response = await fetch(`${baseUrl}/sessions/share`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        messages,
+        description: description || 'Shared Session',
+        base_url: baseUrl,
+      }),
+    });
```

Comment:
just a general comment, in a follow up could be worth it to enable `Content-Encoding` for efficiency since this is all text

Reply:
this makes sense. the other thing to check would be: max size of a DynamoDB item is 400KB so we'd wanna have some guardrails around that